### PR TITLE
Run IT tests with security plugin (#335)

### LINKED ISSUE #1713: [FEATURE] Run integration tests with security plugin enabled
**Is your feature request related to a problem?**
OpenSearch release process includes running plugin integration tests against a cluster with security plugin enabled.

SQL plugin CI does not validate this configuration therefore it's possible for SQL plugin to fail OpenSearch release process to fail.

**What solution would you like?**
SQL Java CI GitHub action to run `:integ-test:integTest` gradle task against OpenSearch cluster with security plugin enabled as well as a cluster with security plugin disabled.

**What alternatives have you considered?**
Someone can do this manually.

**Do you have any additional context?**
OpenSearch release cannot proceed unless all checks pass. When one fails, it's disruptive for everyone.

image

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
### Description
This PR adds a new IT gradle task: `integTestWithSecurity`. It starts a cluster with security plugin installed (it takes latest snapshot), configures cluster, http client for tests and runs one test which required to be run with security plugin.
A new GHA is added which runs this test task.

Please, see team review and discussion in https://github.com/Bit-Quill/opensearch-project-sql/pull/335.
 
### Issues Resolved
fixes #1713
 
### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: integ-test/build.gradle [hunk 1]
```diff
@@ -62,6 +65,82 @@ ext {
...
+            // https is disabled : because `OpenSearchCluster` is hardcoded to validate cluster health by http
+            // refer how IT framework implemented in security plugin and reuse/copy to activate https
+            'plugins.security.ssl.http.enabled' : 'false',
+            'plugins.security.ssl.http.pemcert_filepath' : 'esnode.pem',
+            'plugins.security.ssl.http.pemkey_filepath' : 'esnode-key.pem',
+            'plugins.security.ssl.http.pemtrustedcas_filepath' : 'root-ca.pem',
+            'plugins.security.allow_unsafe_democertificates' : 'true',
+
+            'plugins.security.allow_default_init_securityindex' : 'true',
+            //'plugins.security.authcz.admin_dn' : 'CN=kirk,OU=client,O=client,L=test,C=de',
```
Comment (acarbonetto) [inline integ-test/build.gradle:131]: do we need this?

File: integ-test/build.gradle [hunk 2]
```diff
@@ -108,6 +187,7 @@ dependencies {
     testImplementation group: 'com.h2database', name: 'h2', version: '2.2.220'
     testImplementation group: 'org.xerial', name: 'sqlite-jdbc', version: '3.41.2.2'
     testImplementation group: 'com.google.code.gson', name: 'gson', version: '2.8.9'
+    testCompileOnly 'org.apiguardian:apiguardian-api:1.1.2'
```
Comment (acarbonetto) [inline integ-test/build.gradle:190]: can you explain why this is here?

**Review by acarbonetto — APPROVED** [review]

File: integ-test/src/test/java/org/opensearch/sql/ppl/CrossClusterSearchIT.java [hunk 3]
```diff
@@ -14,30 +14,53 @@
...
 public class CrossClusterSearchIT extends PPLIntegTestCase {
 
-  @Rule public ExpectedException exceptionRule = ExpectedException.none();
+  static {
+    // find a remote cluster
+    String[] clusterNames = System.getProperty("cluster.names").split(",");
+    var remote = "remoteCluster";
+    for (var cluster : clusterNames) {
+      if (cluster.startsWith("remote")) {
+        remote = cluster;
```
Comment (acarbonetto) [inline integ-test/src/test/java/org/opensearch/sql/ppl/CrossClusterSearchIT.java:31]: break;

**Review by acarbonetto — APPROVED** [review]

File: integ-test/build.gradle (hunk 1 above)
Comment (Yury-Fridlyand) [reply integ-test/build.gradle:131]: This is a part of demo configuration, I replated it with example from docs. Not sure how/whether it works.

**Comment by seankao-az** [comment]
With this change, can we now safely remove this line or not?
https://github.com/opensearch-project/sql/pull/1986/files#diff-862918fc998b069454d1afe817b72a1765ad64633f8e726ef74d6517f93fda5fL309

**Review by acarbonetto — DISMISSED** [review]

File: integ-test/build.gradle (hunk 1 above)
Comment (cwperks) [reply integ-test/build.gradle:131]: This is the subject of the admin certificate. You most likely will not need it for doing testing using testClusters. 

You would need it if you plan to use securityadmin. Copying the demo config should be fine.

Comment (Yury-Fridlyand) [reply integ-test/build.gradle]: Thank you @cwperks!
I updated config and added a comment in 23c83ec3c.

**Comment by Yury-Fridlyand** [comment]
> With this change, can we now safely remove this line or not? https://github.com/opensearch-project/sql/pull/1986/files#diff-862918fc998b069454d1afe817b72a1765ad64633f8e726ef74d6517f93fda5fL309 Can we add a condition to exclude it only if security plugin is enabled

I moved this test to another directory, so it won't be executed as part of main IT task.
`integTestWithSecurity` runs it with security plugin on every push/PR.

**Review by acarbonetto — APPROVED** [review]

**Review by YANG-DB — APPROVED** [review]

**Comment by vamsi-amazon** [comment]
@Yury-Fridlyand  Will the integTestWithSecurity get triggered during build pipeline steps?

**Comment by Yury-Fridlyand** [comment]
@vamsi-amazon,
By jenkins? No. Should be?

**Comment by vamsi-amazon** [comment]
https://github.com/opensearch-project/opensearch-build/blob/main/scripts/default/integtest.sh
Integ tests in release pipeline are triggered using above script.
Instead of having a separate command, can we use Security Enabled flag.?
Cc: @seankao-az

**Pull request merged**